_studio/mfx_lib/shared/src/libmfxsw.cpp

```diff
@@ -100,7 +100,14 @@ mfxStatus MFXInitEx(mfxInitParam par, mfxSession *session)
     MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_API, "MFXInit");
     MFX_LTRACE_1(MFX_TRACE_LEVEL_API, "^ModuleHandle^libmfx=", "%p", g_hModule);
 
-        // check error(s)
+    // check the library version
+    if ((MFX_VERSION_MAJOR != par.Version.Major) ||
+        (MFX_VERSION_MINOR < par.Version.Minor))
+    {
+        mfxRes = MFX_ERR_UNSUPPORTED;
```

Comment:
Looks like we have to return w/ unsupported here rather than just assign [mfxRes]

Reply:
Yes. My mistake. Thank you.